packages/core/src/pipes/pipes.ts

```diff
@@ -256,6 +261,38 @@ export class Pipe {
 		return currentResponse;
 	}
 
+	private async isLocalServerRunning(): Promise<Boolean> {
+		// TODO: Remove hard-coded port number when configurable support is added.
+		const portInUse = await isPortInUse(9000);
+
+		// Port not in use. BaseAI dev server is not running.
+		if (!portInUse) {
+			console.warn(
```

Comment:
think we need spacing, new line at the top and bottom of this. And it's an error, should be red.